.teamcity/src/main/kotlin/configurations/Gradleception.kt

```diff
@@ -20,25 +26,67 @@ class Gradleception(model: CIBuildModel, stage: Stage) : BaseGradleBuildType(sta
         javaCrash = false
     }
 
+    params {
+        // Override the default commit id so the build steps produce reproducible distribution
+        param("env.BUILD_COMMIT_ID", "HEAD")
+    }
+
+    /*
+     To avoid unnecessary rerun, what we do here is a bit complicated:
+
+     1. Build a Gradle distribution with a fixed timestamp and hash it, but never use this distribution.
+     2. Build a Gradle distribution with this hash as a version + fixed timestamp -> dogfood-first
+     3. Build a Gradle distribution using dogfood-first with the "hash of dogfood-first" as a version + fixed timestamp different from the one above -> dogfood-second
+     4. Use dogfood-second to run a Gradle build which runs tests.
+
+     */
+    val dogfoodTimestamp1 = "20210101010101+0000"
```

Comment:
Shall we put the timestamps even further in the past (1980?), so the distribution we build here won't be accidentally used?

Reply:
I was no sure `1980` can be handled properly as a version. Let's try it.